etc/change_stream_legacy/src/main.rs

```diff
@@ -0,0 +1,416 @@
+                    }]
+                }),
+            }
+        }
+    }
+
+    fn ys<S: Into<String>>(s: S) -> Value {
+        Value::String(s.into())
+    }
+
+    const GLOBAL_DB1_NAME: &str = "globalDatabase0";
+    const GLOBAL_DB2_NAME: &str = "globalDatabase1";
+    const GLOBAL_COLL1_NAME: &str = "globalCollection0";
+    const GLOBAL_COLL2_NAME: &str = "globalCollection1";
+    const GLOBAL_DB2_COLL1_NAME: &str = "globalDb1Collection0";
+    const GLOBAL_DB1_COLL2_NAME: &str = "globalDb0Collection1";
+
+    fn parse_operation(file: &legacy::File, old: legacy::Operation) -> Operation {
+        let object = {
+            let coll_num = if old.collection == file.collection_name {
+                1
+            } else if Some(&old.collection) == file.collection2_name.as_ref() {
+                2
+            } else {
+                panic!("unexpected collection name {:?}", old.collection);
+            };
+            let db_num = if old.database == file.database_name {
+                1
+            } else if Some(&old.database) == file.database2_name.as_ref() {
+                2
+            } else {
+                panic!("unexpected database name {:?}", old.database);
+            };
+            match (db_num, coll_num) {
+                (1, 1) => GLOBAL_COLL1_NAME,
+                (1, 2) => GLOBAL_DB1_COLL2_NAME,
+                (2, 1) => GLOBAL_DB2_COLL1_NAME,
+                (2, 2) => GLOBAL_DB2_NAME,
+                _ => panic!("invalid target {:?}", (db_num, coll_num)),
+            }
+        }
+        .to_string();
+        let mut arguments = old.arguments;
+        if let Some(args) = &mut arguments {
+            fix_names(file, args);
+        }
+        Operation {
+            name: old.name,
+            object,
+            arguments,
+            ..Operation::default()
+        }
+    }
+
+    fn fix_names(file: &legacy::File, map: &mut serde_yaml::Mapping) {
+        visit_mut(map, &|_, val| match val {
+            Value::String(s) => {
+                if s == &file.database_name {
+                    *s = DB1_NAME.to_string()
+                } else if Some(&*s) == file.database2_name.as_ref() {
+                    *s = DB2_NAME.to_string()
+                } else if s == &file.collection_name {
+                    *s = COLL1_NAME.to_string()
+                } else if Some(&*s) == file.collection2_name.as_ref() {
+                    *s = COLL2_NAME.to_string()
+                }
+            }
+            _ => (),
+        });
+    }
+
+    fn fix_placeholders(map: &mut serde_yaml::Mapping) {
+        visit_mut(map, &|_, val| match val {
+            Value::Number(num) if num.as_i64() == Some(42) => *val = exists(),
+            Value::String(s) if s == "42" => *val = exists(),
+            _ => (),
+        });
+    }
+
+    fn parse_success(file: &legacy::File, mut suc: serde_yaml::Mapping) -> Operation {
+        visit_mut(&mut suc, &|key, val| {
+            if key == "fullDocument" {
```

Comment:
Why is this extra step for `fullDocument` necessary? I noticed it adds an assertion that didn't always exist in the legacy version (e.g. in "$changeStream must be the first stage in a change stream pipeline sent to the server").

Reply:
The legacy test matching was subset at all levels, whereas the unified is more strict and doesn't allow unexpected keys in most places.  Practically, what this meant is that doing the test without adding some `{ $$exists: true }` clauses would fail because the server output had fields not listed in the test.  The `fullDocument._id` case showed up in most of the tests, so it was worth automating.